src/coprocessor/statistics/analyze.rs

```diff
@@ -77,23 +85,44 @@ impl<S: Snapshot> AnalyzeContext<S> {
     // it would build a histogram and count-min sketch of index values.
     fn handle_index(
```

Comment:
I think this function has the same blocking runtime issue as https://github.com/tikv/tikv/pull/8215/files#r455724514.

Reply:
You may need to re-implement batch index scan...

Reply:
Yes! But the old code here uses the old execution framework which does have this problem too. As @breeswish said, maybe we should not modify the existing executor to support `Analyze`.